File: 412. Fizz Buzz/solution.js
/**
 * @param {number} n
 * @return {string[]}
 */
var fizzBuzz = function(n) {
    let answer = []
    for (let i = 1; i <= n; i++) {
        let a = i % 3 == 0
        let b = i % 5 == 0

        let map = {
            3: "Fizz",
            5: "Buzz"
        }

        let str = ""
        if (a) {
            str += map[3];
        } 
        if (b) {
            str += map[5];
        } 
        if (str.length == 0) {
            str += i.toString();
        }

        answer.push(str)
    }
    return answer;
};

File: 412. Fizz Buzz/solution.py
class Solution:
    def fizzBuzz(self, n: int) -> List[str]:
        answer = []
        for i in range(1, n + 1):

            a = i % 3 == 0
            b = i % 5 == 0

            if a and b:
                answer.append("FizzBuzz")
            elif a:
                answer.append("Fizz")
            elif b:
                answer.append("Buzz")
            else:
                answer.append(str(i))
        return answer